### PR TITLE
Fix decay restarts

### PR DESCRIPTION
**Pull request by rgknox**
### Description:

This is a bugfix. These changes are required to ensure b4b comparisons between a base and its restart.  This is due to a recent regression from a very recent tag.

### Expectation of Answer Changes:

This will change the answers following a restart, particularly those answers related to soil carbon, as this will modify the litter flux slightly.  

### Checklist:
- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [x] FATES PASS/FAIL regression tests were run

### Test Results:

TBD

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

**Review by ckoven — APPROVED** [review]
this looks good to me, thanks @rgknox for catching.

**Comment by rgknox** [comment]
Tests on cheyenne:

/glade/scratch/rgknox/clmed-tests/fix-decay-ctsm5.1.dev041

One noted difference, the PRT2 test generated very small differences with the base, see here:

/glade/scratch/rgknox/ERS_D_Ld30.f45_f45_mg37.I2000Clm50FatesCru.cheyenne_intel.clm-FatesPRT2.GC.20210618_075109_ll9h81/run/ERS_D_Ld30.f45_f45_mg37.I2000Clm50FatesCru.cheyenne_intel.clm-FatesPRT2.GC.20210618_075109_ll9h81.clm2.h0.2000-01-01-00000.nc.cprnc.out

The largest rms difference was on the order of e-21.  This might be fine, but I'd like to think through this a little more.

File: biogeochem/FatesSoilBGCFluxMod.F90 [hunk 1]
```diff
@@ -997,9 +997,9 @@ subroutine FluxIntoLitterPools(csite, bc_in, bc_out)
        
        select case (element_list(el))
        case (carbon12_element)
-          bc_out%litt_flux_cel_c_si(:) = 0._r8
-          bc_out%litt_flux_lig_c_si(:) = 0._r8
-          bc_out%litt_flux_lab_c_si(:) = 0._r8
+          bc_out%litt_flux_cel_c_si(:) = 0.0_r8
+          bc_out%litt_flux_lig_c_si(:) = 0.0_r8
+          bc_out%litt_flux_lab_c_si(:) = 0.0_r8
```
Comment (rosiealice) [inline biogeochem/FatesSoilBGCFluxMod.F90:1002]: Embarrassing question I've never got my head around. Does this matter? Like, are there circumstances where you get a different answer with a 0.0_r8 than a 0._r8?

File: main/FatesRestartInterfaceMod.F90 [hunk 2]
```diff
@@ -944,6 +946,17 @@ subroutine define_restart_vars(this, initialize_variables)
            units='kg/m2', veclength=num_elements, flushval = flushzero, &
            hlms='CLM:ALM', initialize=initialize_variables, ivar=ivar, index = ir_seedgerm_litt)
 
+
+    call this%RegisterCohortVector(symbol_base='fates_seed_decay', vtype=cohort_r8, &
+            long_name_base='seed bank (non-germinated)',  &
```
Comment (rosiealice) [inline main/FatesRestartInterfaceMod.F90:951]: Is this 'long name' accurate? Like, if it's the seed decay rate, it's not actually in the seed bank any more?

File: main/FatesRestartInterfaceMod.F90 [hunk 3]
```diff
@@ -944,6 +946,17 @@ subroutine define_restart_vars(this, initialize_variables)
...
+            units='kg/m2', veclength=num_elements, flushval = flushzero, &
+            hlms='CLM:ALM', initialize=initialize_variables, ivar=ivar, index = ir_seed_decay_litt)
+
+    call this%RegisterCohortVector(symbol_base='fates_seedgerm_decay', vtype=cohort_r8, &
+           long_name_base='seed bank (germinated)',  &
+           units='kg/m2', veclength=num_elements, flushval = flushzero, &
```
Comment (rosiealice) [inline main/FatesRestartInterfaceMod.F90:957]: Relatedly, while I know this isn't really related to this bug fix, without delving into the code a bit more, I don't think it's super clear from the names how these variables differ.

**Review by rosiealice — APPROVED** [review]
Comments here entirely related to my own slight incomprehension of how this code has evolved. Otherwise it looks fine.

File: biogeochem/FatesSoilBGCFluxMod.F90 (hunk 1 above)
Comment (rgknox) [reply biogeochem/FatesSoilBGCFluxMod.F90:1002]: nope, I don't think it matters.  This was just an artifact too, I didn't make the change for stylistic reasons.  This is what happened.  I was testing the code by forcing a non zero decomposition flux into E3SM with a value of 1.0. I wanted to see if E3SM was closing its carbon budget with a constant and known input flux from FATES. When I reverted it, I just added a zero.

File: main/FatesRestartInterfaceMod.F90 (hunk 2 above)
Comment (rgknox) [reply main/FatesRestartInterfaceMod.F90:951]: good catch, that is not the right long name. Its a flux rate. I stole the long_name from the pool variable that was already there and forgot to update the name.  Will do.

File: main/FatesRestartInterfaceMod.F90 (hunk 3 above)
Comment (rgknox) [reply main/FatesRestartInterfaceMod.F90:957]: will update this as well, thanks

**Comment by rgknox** [comment]
I reverted a portion of the code where the labile fluxes to the HLM were calculated. The diagnostic site level array with effluxes is populated after the restart when we set the fragmentation fluxes, so it should work. I reverted it to use the cohort level effluxes, to preserve b4b with the test which seems safer.  I think the difference could be explained by order of operations rounding errors (which are acceptable).

**Comment by rgknox** [comment]
retesting

**Comment by rgknox** [comment]
tests pass: /glade/scratch/rgknox/clmed-tests/fix-decay-ctsm5.1.dev041-v4

**Pull request merged**